Using element only on Atlantic V3

Hi,

I have installed the AquaMQTT module.  Everything seems to work, but for some reasons i am not able to use the element only modus. I have setup the following as mentioned in the FAQ:
[https://tweakers.net/fotoalbum/image/PNsoNaETxv1jHAUhK6jbrbxD.png](url)

Am i doing something wrong? Is there anyone that is using a V3 Atlantic Explorer with the same problems? The AquaMQTT heating element stays ''OFF"

A friend of my tried, and this is the output:

`src/message/next/HMIMessage.cpp:7:1: error: no declaration matches 'aquamqtt::message::next::HMIMessage::HMIMessage(uint8_t*, const uint8_t*)'
    7 | HMIMessage::HMIMessage(uint8_t* data, const uint8_t* previous) : mData(data)
      | ^~~~
In file included from src/message/next/HMIMessage.cpp:1:
include/message/next/HMIMessage.h:11:7: note: candidates are: 'aquamqtt::message::next::HMIMessage::HMIMessage(const aquamqtt::message::next::HMIMessage&)'
   11 | class HMIMessage : public IHMIMessage
      |       ^~~~
include/message/next/HMIMessage.h:14:14: note:                 'aquamqtt::message::next::HMIMessage::HMIMessage(uint8_t*, uint8_t*)'
   14 |     explicit HMIMessage(uint8_t* data, uint8_t* previous = nullptr);
      |              ^~~~
include/message/next/HMIMessage.h:11:7: note: 'class aquamqtt::message::next::HMIMessage' defined here
   11 | class HMIMessage : public IHMIMessage
      |       ^~~~
src/message/next/HMIMessage.cpp:13:6: error: no declaration matches 'void aquamqtt::message::next::HMIMessage::compareWith(const uint8_t*)'
   13 | void HMIMessage::compareWith(const uint8_t* data)
      |      ^~~~
include/message/next/HMIMessage.h:46:10: note: candidate is: 'void aquamqtt::message::next::HMIMessage::compareWith(uint8_t*)'
   46 |     void compareWith(uint8_t* data);
      |          ^~~~~
include/message/next/HMIMessage.h:11:7: note: 'class aquamqtt::message::next::HMIMessage' defined here
   11 | class HMIMessage : public IHMIMessage
      |       ^~~~
src/message/next/HMIMessage.cpp: In member function 'virtual uint8_t aquamqtt::message::next::HMIMessage::getAttr(aquamqtt::message::HMI_ATTR_U8)':
src/message/next/HMIMessage.cpp:185:24: error: 'OT_ALWAYS_ON' was not declared in this scope; did you mean 'ALWAYS_ON'? 
  185 |                 return OT_ALWAYS_ON;
      |                        ^~~~~~
      |                        ALWAYS_ON
src/message/next/HMIMessage.cpp:187:20: error: 'OT_TIMER' was not declared in this scope; did you mean 'TIMER'?
  187 |             return OT_TIMER;
      |                    ^~~~
      |                    TIMER
src/message/next/HMIMessage.cpp: In member function 'virtual void aquamqtt::message::next::HMIMessage::setAttr(aquamqtt::message::HMI_ATTR_U8, uint8_t)':
src/message/next/HMIMessage.cpp:386:34: error: 'OT_TIMER' was not declared in this scope; did you mean 'TIMER'?
  386 |             if (operationType == OT_TIMER)
      |                                  ^~~~
      |                                  TIMER
src/message/next/HMIMessage.cpp:390:39: error: 'OT_ALWAYS_ON' was not declared in this scope; did you mean 'ALWAYS_ON'? 
  390 |             else if (operationType == OT_ALWAYS_ON)
      |                                       ^~~~~~
      |                                       ALWAYS_ON
* [.pio\build\esp32-s3-devkitc-1\src\message\next\HMIMessage.cpp.o] Error 1`

--- a/AquaMQTT/include/Version.h
+++ b/AquaMQTT/include/Version.h
@@ -1,4 +1,4 @@
 namespace aquamqtt
 {
-constexpr char VERSION[] = "v1.7.0";
+constexpr char VERSION[] = "v1.7.1";
 }

--- a/AquaMQTT/src/message/next/HMIMessage.cpp
+++ b/AquaMQTT/src/message/next/HMIMessage.cpp
@@ -217,7 +217,7 @@ bool HMIMessage::getAttr(const HMI_ATTR_BOOL attr)
         case HMI_ATTR_BOOL::EMERGENCY_MODE_ENABLED:
             return mData[5] & 0x01;
         case HMI_ATTR_BOOL::HEATING_ELEMENT_ALLOWED:
-            return mData[8] & 0x04;
+            return !(mData[8] & 0x04);
         case HMI_ATTR_BOOL::PV_INPUT_ALLOWED:
             return mData[8] & 0x02;
     }
@@ -423,12 +423,12 @@ void HMIMessage::setAttr(const HMI_ATTR_BOOL attr, const bool value)
         case HMI_ATTR_BOOL::HEATING_ELEMENT_ALLOWED:
             if (value)
             {
-                mData[8] |= 0x04;
+                mData[8] &= ~0x04;
             }
             // Sanity: You cannot disable heating element if emergency mode is activated
             else if (!getAttr(HMI_ATTR_BOOL::EMERGENCY_MODE_ENABLED))
             {
-                mData[8] &= ~0x04;
+                mData[8] |= 0x04;
             }
             break;
         case HMI_ATTR_BOOL::PV_INPUT_ALLOWED:

